TestTCP fails on Mac, prevents build completion

#### Problem
After #2689 landed, when I run `./gn_build.sh` I get a log like so:

```
'#0:','Test-CHIP-Tcp'
'#2:','Setup                 ','PASSED'
'#3:','Simple Init Test IPV4 ','PASSED'
CHIP node ready to service events; PID: 13071; PPID: 13066
Connection closed.Connection complete encountered an error: OS Error 2006 (0x000007D6): Device not configuredSO_LINGER: 22Connection closed.Freeing closed connection.Failed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in useFailed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in useAsync DNS worker thread woke up.Async DNS worker thread exiting.Async DNS worker thread woke up.Async DNS worker thread exiting.../../src/transport/tests/TestTCP.cpp:138: assertion failed: "ReceiveHandlerCallCount == 1"
'#3:','Message Self Test IPV4','FAILED'
../../src/transport/tests/TestTCP.cpp:85: assertion failed: "err == CHIP_NO_ERROR"
'#3:','Simple Init Test IPV6 ','FAILED'
../../src/transport/tests/TestTCP.cpp:117: assertion failed: "err == CHIP_NO_ERROR"
../../src/transport/tests/TestTCP.cpp:135: assertion failed: "err == CHIP_NO_ERROR"
../../src/transport/tests/TestTCP.cpp:138: assertion failed: "ReceiveHandlerCallCount == 1"
'#3:','Message Self Test IPV6','FAILED'
'#4:','Teardown              ','PASSED'
'#6:','3','4'
'#7:','5','8'
INF Test 1/1: [FAIL] TestTCP
[36/58] ACTION //src/transport/tests:TestTCP_run(//build/toolchain/host:mac_x64_gcc)
FAILED: mac_x64_gcc/gen/src/transport/tests/TestTCP_run.pw_pystamp 
python ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /Users/bzbarsky/connectedhomeip/ --current-path /Users/bzbarsky/connectedhomeip/src/transport/tests --default-toolchain=//third_party/pigweed/repo/pw_toolchain/dummy:dummy --current-toolchain=//build/toolchain/host:mac_x64_gcc --touch /Users/bzbarsky/connectedhomeip/out/debug/mac_x64_gcc/gen/src/transport/tests/TestTCP_run.pw_pystamp --capture-output -- /Users/bzbarsky/connectedhomeip/third_party/pigweed/repo/pw_unit_test/py/pw_unit_test/test_runner.py --runner ../../third_party/pigweed/repo/targets/host/run_test --test mac_x64_gcc/tests/TestTCP
INF Test 1/1: [ RUN] TestTCP
ERR ../../third_party/pigweed/repo/targets/host/run_test exited with status 252
OUT [13083]
Failed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in useFailed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in use'#0:','Test-CHIP-Tcp'
'#2:','Setup                 ','PASSED'
../../src/transport/tests/TestTCP.cpp:85: assertion failed: "err == CHIP_NO_ERROR"
'#3:','Simple Init Test IPV4 ','FAILED'
../../src/transport/tests/TestTCP.cpp:117: assertion failed: "err == CHIP_NO_ERROR"
../../src/transport/tests/TestTCP.cpp:135: assertion failed: "err == CHIP_NO_ERROR"
CHIP node ready to service events; PID: 13089; PPID: 13083
Failed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in useFailed to initialize TCP transport: OS Error 2048 (0x00000800): Address already in useAsync DNS worker thread woke up.Async DNS worker thread exiting.Async DNS worker thread woke up.Async DNS worker thread exiting.../../src/transport/tests/TestTCP.cpp:138: assertion failed: "ReceiveHandlerCallCount == 1"
'#3:','Message Self Test IPV4','FAILED'
../../src/transport/tests/TestTCP.cpp:85: assertion failed: "err == CHIP_NO_ERROR"
'#3:','Simple Init Test IPV6 ','FAILED'
../../src/transport/tests/TestTCP.cpp:117: assertion failed: "err == CHIP_NO_ERROR"
../../src/transport/tests/TestTCP.cpp:135: assertion failed: "err == CHIP_NO_ERROR"
../../src/transport/tests/TestTCP.cpp:138: assertion failed: "ReceiveHandlerCallCount == 1"
'#3:','Message Self Test IPV6','FAILED'
'#4:','Teardown              ','PASSED'
'#6:','4','4'
'#7:','8','8'
INF Test 1/1: [FAIL] TestTCP
ninja: build stopped: subcommand failed.
```

 #### Proposed Solution
Fix the test to actually work.  And probably add newlines to some of that logging that is obviously missing them....

@andy31415

--- a/src/inet/TCPEndPointImplSockets.cpp
+++ b/src/inet/TCPEndPointImplSockets.cpp
@@ -1134,7 +1134,12 @@ CHIP_ERROR TCPEndPointImplSockets::CheckConnectionProgress(bool & isProgressing)
 
     // Fetch the bytes pending successful transmission in the TCP out queue.
 
+#ifdef __APPLE__
+    socklen_t len = sizeof(currPendingBytesRaw);
+    if (getsockopt(mSocket, SOL_SOCKET, SO_NWRITE, &currPendingBytesRaw, &len) < 0)
+#else
     if (ioctl(mSocket, TIOCOUTQ, &currPendingBytesRaw) < 0)
+#endif
     {
         return CHIP_ERROR_POSIX(errno);
     }

--- a/src/transport/raw/tests/BUILD.gn
+++ b/src/transport/raw/tests/BUILD.gn
@@ -42,11 +42,8 @@ chip_test_suite("tests") {
   test_sources = [
     "TestMessageHeader.cpp",
     "TestPeerAddress.cpp",
+    "TestTCP.cpp",
   ]
-
-  if (current_os != "mac") {
-    test_sources += [ "TestTCP.cpp" ]
-  }
 
   public_deps = [
     ":helpers",

